iothub/device/src/IotHubBaseClient.cs

```diff
@@ -60,14 +63,28 @@ public abstract class IotHubBaseClient : IAsyncDisposable
                 ModelId = _clientOptions.ModelId,
                 PayloadConvention = _clientOptions.PayloadConvention,
                 IotHubClientTransportSettings = _clientOptions.TransportSettings,
-                HttpOperationTransportSettings = _clientOptions.FileUploadTransportSettings,
                 MethodCallback = OnMethodCalledAsync,
                 DesiredPropertyUpdateCallback = OnDesiredStatePatchReceived,
                 ConnectionStatusChangeHandler = OnConnectionStatusChanged,
                 MessageEventCallback = OnMessageReceivedAsync,
                 RetryPolicy = _clientOptions.RetryPolicy ?? new IotHubClientNoRetry(),
             };
 
+            if (IotHubConnectionCredentials.ModuleId.IsNullOrWhiteSpace())
+            {
+                // Set up file upload settings over HTTP for the device client
+                PipelineContext.HttpOperationTransportSettings = _clientOptions.FileUploadTransportSettings;
```

Comment:
Should we throw on init if client is module and FileUploadTransportSettings are set?

Reply:
We set a new instance by default, so we won't be able to do a check for this:
https://github.com/Azure/azure-iot-sdk-csharp/blob/7091f4e2262753a937493e23dd91fa8a818ffdee/iothub/device/src/IotHubClientOptions.cs#L46